awx/api/views.py

```diff
@@ -608,6 +609,43 @@ class ScheduleDetail(RetrieveUpdateDestroyAPIView):
     new_in_148 = True
 
 
+class SchedulePreview(GenericAPIView):
+
+    model = Schedule
+    view_name = _('Schedule Recurrence Rule Preview')
+    serializer_class = SchedulePreviewSerializer
+    new_in_api_v2 = True
+
+    def post(self, request):
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            next_stamp = now()
+            schedule = []
+            gen = Schedule.rrulestr(serializer.validated_data['rrule']).xafter(next_stamp, count=20)
+
+            # loop across the entire generator and grab the first 10 events
+            for event in gen:
+                if len(schedule) >= 10:
```

Comment:
We need more than 10 events from the generator (20 should probably suffice) in case there's a few "imaginary dates", e.g., "the night of DST transition between the hours of 2AM and 2:30AM"